autoscale/src/main/java/com/sequenceiq/periscope/monitor/CleanupMonitor.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.periscope.monitor;
+
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_FAILED;
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_SUCCESS;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.api.model.ActivityStatus;
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.cleanup.ActivityCleanupEvaluator;
+
+@Component
+@ConditionalOnProperty(prefix = "periscope.enabledAutoscaleMonitors.cleanup-monitor", name = "enabled", havingValue = "true")
+public class CleanupMonitor extends ScalingActivityMonitor {
+
+    @Override
+    protected List<ScalingActivities> getMonitored() {
+        Set<ActivityStatus> statuses = EnumSet.of(SCALING_FLOW_FAILED, SCALING_FLOW_SUCCESS);
+        Set<Long> activityIdsToCleanup = getScalingActivityService().findAllInStatusesThatStartedBefore(statuses, getCleanupConfig().getCleanupDurationHours());
+        if (activityIdsToCleanup.isEmpty()) {
+            return Collections.emptyList();
+        }
+        // TODO: An ID for this monitored list must be generated here, it always defaults to 0
```

Comment:
What's the implication of this defaulting to 0? The task the is scheduled will replace any other cleanup-task which is somehow stuck in the backlog?

If there's nothing to be done here - better to replace the TODO with the implications/reason for the special case.

Reply:
Currently, the RejectedThreadMonitor caters to only threads associated with periscope clusters. Rejected threads are mapped by the clusterId as the key in a concurrent hashmap. Here, the monitored data (`ScalingActivities`) is a collection of activity Ids that do not belong to a particular cluster but to multiple clusters. Since each thread is not associated with a single cluster but with multiple scaling activities, handling rejected threads will not work as intended